src/backend/executor/execMain.c

```diff
@@ -795,7 +796,10 @@ standard_ExecutorRun(QueryDesc *queryDesc,
     {
         if (Gp_role == GP_ROLE_EXECUTE ||
             sliceRunsOnQD(currentSlice))
-            currentSliceId = currentSlice->sliceIndex;
+			{
+            	currentSliceId = currentSlice->sliceIndex;
+				amIParallel = currentSlice->useMppParallelMode;
```

Comment:
need to and  with `parallelModeNeeded ` ?

Reply:
No need, useMppParallelMode is set if slice's parallel_workers > 0.